src/js/modules/infragistics.datasource.js

```diff
@@ -5493,6 +5497,146 @@
+				search = search.replace(matches[ i ], "");
+				res.push(matches[ i ].replace(/\"/g, ""));
+			}
+			matches = search.split(/\s+/);
+			res = res.concat(matches);
+			return res;
+		},
+		_filterAllFields: function (val, data, fields) {
+			fields = fields || this.schema().fields();
+			if (!fields || !fields.length) {
+				return [];
+			}
+			var i, j, len = data.length,
+				filteredData = [], count = 0, curr,
+				formattedRecords = (this.schema() || {})._formattedRecords || [],
+				push,
+				searchExprs = this._splitFilterExpression(val), se = [];/*split by spaces */
+			for (i = 0; i < searchExprs.length; i++) {
+				curr = searchExprs[ i ];
+				if (curr) {// remove empty strings
+					push = true;
+					/* check whether there are duplicate tokens and remove those that are substring tokens in array "se"
+					For instance if array "searchExpr" is ["1", "2", "123"], then array "se" should be ["123"]*/
+					for (j = 0; j < se.length; j++) {
+						if (se[ j ] === curr || se[ j ].indexOf(curr) !== -1) {
+							push = false;
+							break;
+						}
+						/* remove those elements in array "se" which are substring of variable "curr" */
+						if (curr.indexOf(se[ j ]) !== -1) {
+							se.splice(j, 1);
+							j = -1;
+						}
+					}
+					if (push) {
+						se.push(this.settings.filtering.caseSensitive ? curr : curr.toLowerCase());
+					}
+				}
+			}
+			if (!se.length) {
+				return data;
+			}
+			for (i = 0; i < len; i++) {
+				if (this._findMatchByFields(se, data[ i ], fields, formattedRecords[ i ] || null)) {
+					filteredData[ count++ ] = data[ i ];
+				}
+			}
+			return filteredData;
+		},
+		_findMatchByFields: function (searchTokens, rec, fields, formattedRecord) {
+			var j, fl = fields.length, dataVal, fieldName, i,
+				ignoreCase = !this.settings.filtering.caseSensitive, s = "";
+			for (j = 0; j < fl; j++) {
+				fieldName = fields[ j ].name;
+				if (formattedRecord && formattedRecord[ fieldName ] !== undefined) {
+					dataVal = formattedRecord[ fieldName ];
+				} else {
+					dataVal = this._hasMapper ? this.getCellValue(fieldName, rec) : rec[ fieldName ];
+				}
+				dataVal = (dataVal === null || dataVal === undefined) ? "" : dataVal;
+				s += dataVal + "||";
+			}
+			s = ignoreCase ? s.toLowerCase() : s;
+			for (i = 0; i < searchTokens.length; i++) {
+				if (s.indexOf(searchTokens[ i ]) === -1) {
+					return false;
+				}
+			}
+			return true;
+		},
+		_getFieldsWithFormatter: function () {
+			/* return array of fields(from the data schema) that have property "formatter" defined
+			if schema is not defined OR there aren't such fields - ruturns empty array */
+			var i, f, res = [];
+			if (!this.schema()) {
+				return res;
+			}
+			f = this.schema().fields();
+			for (i = 0; i < f.length ; i++) {
+				if (f[ i ].formatter) {
+					res.push(f[ i ]);
+				}
+			}
+			return res;
+		},
+		_generateFormattedRecords: function (data) {
+			var i, j, f, len, fr = [], schema = this.schema(),
```

Comment:
Since currently this is implementation on DataSource level and not on Grid level similar scenarios are not supported. 
If the users sets the field formatter though they will need to handle it themselves since we cannot predict what the user would use.